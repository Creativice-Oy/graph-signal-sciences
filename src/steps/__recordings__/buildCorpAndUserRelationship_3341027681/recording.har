{
  "log": {
    "_recordingName": "buildCorpAndUserRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "a0d858918dc7000c5fe61c66948ecc2e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "x-api-user",
              "value": "[REDACTED]"
            },
            {
              "name": "x-api-token",
              "value": "[REDACTED]"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 228,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps"
        },
        "response": {
          "bodySize": 435,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 435,
            "text": "{\"data\":[{\"name\":\"jupiterone\",\"displayName\":\"JupiterOne\",\"smallIconURI\":\"\",\"created\":\"2022-01-10T18:29:55Z\",\"siteLimit\":5,\"sites\":{\"uri\":\"/api/v0/corps/jupiterone/sites\"},\"authType\":\"builtin\",\"logoutURI\":\"\",\"samlCert\":\"\",\"samlEndpoint\":\"\",\"signRequestsUsingStoredCert\":false,\"samlRequestCert\":\"\",\"sessionMaxAgeDashboard\":2592000,\"apiAllowedIPs\":[],\"apiTokenMaxAge\":0,\"restrictedAccessTokens\":false,\"ssoProvisioningConfigured\":false}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 20 Jan 2022 23:15:58 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "435"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "61e9ed2e03119a01e62de458"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 395,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-20T23:15:57.746Z",
        "time": 396,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 396
        }
      },
      {
        "_id": "a1ad97198eead933a970472a10fd77f9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "x-api-user",
              "value": "[REDACTED]"
            },
            {
              "name": "x-api-token",
              "value": "[REDACTED]"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 245,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps/jupiterone/users"
        },
        "response": {
          "bodySize": 2181,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 2181,
            "text": "{\"data\":[{\"name\":\"Samuel Poulton Observer\",\"email\":\"samuel.poulton+sigsci_observer@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/samuel.poulton+sigsci_observer@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"observer\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-19T21:49:48.458Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false},{\"name\":\"Samuel Poulton\",\"email\":\"samuel.poulton@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/samuel.poulton@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"owner\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-18T17:51:41.435Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false},{\"name\":\"Sam Andrus\",\"email\":\"sam.andrus@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/sam.andrus@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"owner\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-10T18:30:40.62Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false},{\"name\":\"Austin Kelleher\",\"email\":\"austin.kelleher@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/austin.kelleher@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"owner\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-10T18:30:28.296Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 20 Jan 2022 23:15:58 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "61e9ed2e09ae7301e7b3362f"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 402,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-20T23:15:58.172Z",
        "time": 180,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 180
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
