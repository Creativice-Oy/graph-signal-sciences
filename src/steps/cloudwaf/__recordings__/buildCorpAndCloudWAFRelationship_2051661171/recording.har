{
  "log": {
    "_recordingName": "buildCorpAndCloudWAFRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "79c71d206a4c9cf236c090578b230d37",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 64,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "content-length",
              "value": 64
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 240,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "email=sam.andrus%40jupiterone.com&password=DXW4mwz%40vab*gkd1mqf"
          },
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/auth"
        },
        "response": {
          "bodySize": 49,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 49,
            "text": "{\"token\":\"475f59f2-966e-4bfe-bb5e-713854750fda\"}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 07 Nov 2022 13:51:56 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "49"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "63690d7cdafb4ecfcbae4910"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 394,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-11-07T13:51:56.019Z",
        "time": 834,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 834
        }
      },
      {
        "_id": "556a3fcc8d26abfc35cb34522cd84237",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 260,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps/jupiterone/cloudwafInstances"
        },
        "response": {
          "bodySize": 1661,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1661,
            "text": "{\"data\":[{\"id\":\"9adf79f2-6553-4008-a012-e2e78d1c40eb\",\"name\":\"testinstance\",\"description\":\"testinstance\",\"region\":\"us-east-1\",\"tlsMinVersion\":\"1.2\",\"workspaceConfigs\":[{\"siteName\":\"test\",\"instanceLocation\":\"direct\",\"clientIPHeader\":\"\",\"listenerProtocols\":[\"https\"],\"routes\":[{\"certificateIds\":[],\"domains\":[\"paradox-test.blogspot.com\"],\"origin\":\"https://paradox-test.blogspot.com/\",\"passHostHeader\":false,\"id\":\"4d14d79e-1f1b-4c8f-a662-1c07ff4c46db\",\"connectionPooling\":true,\"trustProxyHeaders\":false}]}],\"deployment\":{\"status\":\"done\",\"message\":\"\",\"egressIPs\":[{\"ip\":\"54.157.9.221\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.211.133.174\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"44.198.49.148\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.222.240.59\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"44.198.76.176\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"34.192.82.71\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.216.58.131\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"18.214.162.196\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"54.84.118.7\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"52.204.194.140\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.232.125.243\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"107.22.39.13\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"54.147.50.184\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"54.160.112.153\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.227.47.195\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.208.110.204\",\"status\":\"unknown\",\"updatedAt\":\"\"}],\"dnsEntry\":\"9adf79f2-6553-4008-a012-e2e78d1c40eb.sigscicloudwaf.com\"},\"useUploadedCertificates\":false,\"createdBy\":\"sam.andrus@jupiterone.com\",\"created\":\"2022-11-07T05:27:11Z\",\"updatedBy\":\"\"}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 07 Nov 2022 13:51:57 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "1661"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "63690d7d44e822837ef399e3"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 396,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-11-07T13:51:56.858Z",
        "time": 802,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 802
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
