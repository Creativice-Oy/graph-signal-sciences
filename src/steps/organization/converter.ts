import {
  createIntegrationEntity,
  Entity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';
import { SignalSciencesCorp } from '../../types';

import { Entities } from '../constants';

export function createOrganizationEntity(corp: SignalSciencesCorp): Entity {
  return createIntegrationEntity({
    entityData: {
      source: corp,
      assign: {
        _key: corp.name,
        _type: Entities.ORGANIZATION._type,
        _class: Entities.ORGANIZATION._class,
        createdOn: parseTimePropertyValue(corp.created),
        name: corp.name,
        displayName: corp.displayName,
        smallIconURI: corp.smallIconURI,
        siteLimit: corp.siteLimit,
        sitesUri: corp.sites?.uri,
        authType: corp.authType,
        logoutURI: corp.logoutURI,
        samlCert: corp.samlCert,
        signRequestsUsingStoredCert: corp.signRequestsUsingStoredCert,
        samlRequestCert: corp.samlRequestCert,
        sessionMaxAgeDashboard: corp.sessionMaxAgeDashboard,
        apiTokenMaxAge: corp.apiTokenMaxAge,
        restrictedAccessTokens: corp.restrictedAccessTokens,
        ssoProvisioningConfigured: corp.ssoProvisioningConfigured,
        webLink: buildWebLink(corp.name),
      },
    },
  });
}

function buildWebLink(corpName) {
  return `https://dashboard.signalsciences.net/corps/${corpName}/overview`;
}
