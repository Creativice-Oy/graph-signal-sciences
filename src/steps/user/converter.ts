import {
  createIntegrationEntity,
  createDirectRelationship,
  parseTimePropertyValue,
  Entity,
  RelationshipClass,
  Relationship,
} from '@jupiterone/integration-sdk-core';

import { Entities } from '../constants';
import { SignalSciencesUser } from '../../types';

export function createUserEntity(
  corpName: string,
  user: SignalSciencesUser,
): Entity {
  return createIntegrationEntity({
    entityData: {
      source: user,
      assign: {
        _type: Entities.USER._type,
        _class: Entities.USER._class,
        _key: user.email,
        createdOn: parseTimePropertyValue(user.created),
        username: user.email,
        name: user.name,
        displayName: user.name,
        email: user.email,
        mfaEnabled: user.mfaEnabled,
        active: user.status === 'active',
        role: user.role,
        webLink: buildWebLink(corpName, user.email),
      },
    },
  });
}

export function createOrganizationUserRelationship(
  orgainzation: Entity,
  user: Entity,
): Relationship {
  return createDirectRelationship({
    _class: RelationshipClass.HAS,
    from: orgainzation,
    to: user,
  });
}

function buildWebLink(corpName: string, userEmail: string) {
  return `https://dashboard.signalsciences.net/corps/${corpName}/users/${userEmail}`;
}
