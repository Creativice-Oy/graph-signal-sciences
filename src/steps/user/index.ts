import {
  IntegrationStep,
  IntegrationStepExecutionContext,
} from '@jupiterone/integration-sdk-core';

import { createAPIClient } from '../../client';
import { IntegrationConfig } from '../../config';
import { Entities, Steps, Relationships } from '../constants';
import {
  createOrganizationUserRelationship,
  createUserEntity,
} from './converter';

export async function fetchUsers({
  instance,
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config);

  await jobState.iterateEntities(
    { _type: Entities.ORGANIZATION._type },
    async (organizationEntity) => {
      const corpName = organizationEntity._key;
      await apiClient.iterateUsers(corpName, async (user) => {
        const userEntity = await jobState.addEntity(
          createUserEntity(corpName, user),
        );
        await jobState.addRelationship(
          createOrganizationUserRelationship(organizationEntity, userEntity),
        );
      });
    },
  );
}

export const userSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: Steps.USERS,
    name: 'Fetch Users',
    entities: [Entities.USER],
    relationships: [Relationships.ORGANIZATION_HAS_USER],
    dependsOn: [Steps.ORGANIZATIONS],
    executionHandler: fetchUsers,
  },
];
