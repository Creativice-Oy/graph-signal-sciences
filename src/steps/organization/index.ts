import {
  IntegrationStep,
  IntegrationStepExecutionContext,
} from '@jupiterone/integration-sdk-core';

import { createAPIClient } from '../../client';
import { IntegrationConfig } from '../../config';
import { Steps, Entities } from '../constants';
import { createOrganizationEntity } from './converter';

export const ORGANIZATIONS_ENTITY_KEY = 'entity:organizations';

export async function fetchCorps({
  logger,
  instance,
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(logger, instance.config);

  logger.info('Querying for corps from signal sciences...');

  await apiClient.iterateCorps(async (corp) => {
    await jobState.addEntity(createOrganizationEntity(corp));
  });
}

export const organizationSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: Steps.ORGANIZATIONS,
    name: 'Fetch Corps',
    entities: [Entities.ORGANIZATION],
    relationships: [],
    dependsOn: [],
    executionHandler: fetchCorps,
  },
];
