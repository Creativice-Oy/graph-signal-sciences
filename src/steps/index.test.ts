import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';

import { setupSigSciRecording } from '../../test/recording';

import { IntegrationConfig } from '../config';
import { fetchUsers } from './user';
import { fetchCorps } from './organization';
import { integrationConfig } from '../../test/config';

describe('should collect corp and user data', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupSigSciRecording({
      directory: __dirname,
      name: 'buildCorpAndUserRelationship',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('validate client integration and converters', async () => {
    // Arrange
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    // Act
    await fetchCorps(context);
    await fetchUsers(context);

    // Assert
    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const accounts = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('Organization'),
    );
    expect(accounts.length).toBeGreaterThan(0);
    expect(accounts).toMatchGraphObjectSchema({
      _class: ['Organization'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'sigsci_corp' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          displayName: { type: 'string' },
          createdOn: { type: 'number' },
          smallIconURI: { type: 'string' },
          siteLimit: { type: 'number' },
          sitesUri: { type: 'string' },
          authType: { type: 'string' },
          logoutURI: { type: 'string' },
          samlCert: { type: 'string' },
          samlEndpoint: { type: 'string' },
          signRequestsUsingStoredCert: { type: 'boolean' },
          samlRequestCert: { type: 'string' },
          sessionMaxAgeDashboard: { type: 'number' },
          apiTokenMaxAge: { type: 'number' },
          restrictedAccessTokens: { type: 'boolean' },
          ssoProvisioningConfigured: { type: 'boolean' },
        },
      },
    });

    const users = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('User'),
    );
    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'sigsci_user' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          displayName: { type: 'string' },
          createdOn: { type: 'number' },
          username: { type: 'string' },
          email: { type: 'string' },
          mfaEnabled: { type: 'boolean' },
          active: { type: 'boolean' },
          role: { type: 'string' },
          webLink: { type: 'string' },
        },
      },
    });
  });
});
