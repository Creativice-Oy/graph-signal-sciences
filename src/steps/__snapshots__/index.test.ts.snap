// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collect corp and user data validate client integration and converters 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Organization",
      ],
      "_key": "jupiterone",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "apiAllowedIPs": Array [],
            "apiTokenMaxAge": 0,
            "authType": "builtin",
            "created": "2022-01-10T18:29:55Z",
            "displayName": "JupiterOne",
            "logoutURI": "",
            "name": "jupiterone",
            "restrictedAccessTokens": false,
            "samlCert": "",
            "samlEndpoint": "",
            "samlRequestCert": "",
            "sessionMaxAgeDashboard": 2592000,
            "signRequestsUsingStoredCert": false,
            "siteLimit": 5,
            "sites": Object {
              "uri": "/api/v0/corps/jupiterone/sites",
            },
            "smallIconURI": "",
            "ssoProvisioningConfigured": false,
          },
        },
      ],
      "_type": "sigsci_corp",
      "apiTokenMaxAge": 0,
      "authType": "builtin",
      "createdOn": 1641839395000,
      "displayName": "JupiterOne",
      "logoutURI": "",
      "name": "jupiterone",
      "restrictedAccessTokens": false,
      "samlCert": "",
      "samlRequestCert": "",
      "sessionMaxAgeDashboard": 2592000,
      "signRequestsUsingStoredCert": false,
      "siteLimit": 5,
      "sitesUri": "/api/v0/corps/jupiterone/sites",
      "smallIconURI": "",
      "ssoProvisioningConfigured": false,
      "webLink": "https://dashboard.signalsciences.net/corps/jupiterone/overview",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "samuel.poulton+sigsci_observer@jupiterone.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "accessTokenAllowed": true,
            "accessTokens": Object {
              "uri": "/api/v0/user/apitokens",
            },
            "announcements": Object {
              "uri": "/api/v0/user/announcements",
            },
            "apiUser": false,
            "authStatus": "none",
            "corpAuthType": "builtin",
            "created": "2022-01-19T21:49:48.458Z",
            "defaultDashboards": Object {
              "uri": "/api/v0/user/defaultDashboards",
            },
            "email": "samuel.poulton+sigsci_observer@jupiterone.com",
            "memberships": Object {
              "data": Array [],
              "uri": "/api/v0/corps/jupiterone/users/samuel.poulton+sigsci_observer@jupiterone.com/memberships",
            },
            "mfaEnabled": false,
            "name": "Samuel Poulton Observer",
            "role": "observer",
            "ssoProvisioned": false,
            "status": "active",
          },
        },
      ],
      "_type": "sigsci_user",
      "active": true,
      "createdOn": 1642628988458,
      "displayName": "Samuel Poulton Observer",
      "email": "samuel.poulton+sigsci_observer@jupiterone.com",
      "mfaEnabled": false,
      "name": "Samuel Poulton Observer",
      "role": "observer",
      "status": "active",
      "username": "samuel.poulton+sigsci_observer@jupiterone.com",
      "webLink": "https://dashboard.signalsciences.net/corps/jupiterone/users/samuel.poulton+sigsci_observer@jupiterone.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "samuel.poulton@jupiterone.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "accessTokenAllowed": true,
            "accessTokens": Object {
              "uri": "/api/v0/user/apitokens",
            },
            "announcements": Object {
              "uri": "/api/v0/user/announcements",
            },
            "apiUser": false,
            "authStatus": "none",
            "corpAuthType": "builtin",
            "created": "2022-01-18T17:51:41.435Z",
            "defaultDashboards": Object {
              "uri": "/api/v0/user/defaultDashboards",
            },
            "email": "samuel.poulton@jupiterone.com",
            "memberships": Object {
              "data": Array [],
              "uri": "/api/v0/corps/jupiterone/users/samuel.poulton@jupiterone.com/memberships",
            },
            "mfaEnabled": false,
            "name": "Samuel Poulton",
            "role": "owner",
            "ssoProvisioned": false,
            "status": "active",
          },
        },
      ],
      "_type": "sigsci_user",
      "active": true,
      "createdOn": 1642528301435,
      "displayName": "Samuel Poulton",
      "email": "samuel.poulton@jupiterone.com",
      "mfaEnabled": false,
      "name": "Samuel Poulton",
      "role": "owner",
      "status": "active",
      "username": "samuel.poulton@jupiterone.com",
      "webLink": "https://dashboard.signalsciences.net/corps/jupiterone/users/samuel.poulton@jupiterone.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sam.andrus@jupiterone.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "accessTokenAllowed": true,
            "accessTokens": Object {
              "uri": "/api/v0/user/apitokens",
            },
            "announcements": Object {
              "uri": "/api/v0/user/announcements",
            },
            "apiUser": false,
            "authStatus": "none",
            "corpAuthType": "builtin",
            "created": "2022-01-10T18:30:40.62Z",
            "defaultDashboards": Object {
              "uri": "/api/v0/user/defaultDashboards",
            },
            "email": "sam.andrus@jupiterone.com",
            "memberships": Object {
              "data": Array [],
              "uri": "/api/v0/corps/jupiterone/users/sam.andrus@jupiterone.com/memberships",
            },
            "mfaEnabled": false,
            "name": "Sam Andrus",
            "role": "owner",
            "ssoProvisioned": false,
            "status": "active",
          },
        },
      ],
      "_type": "sigsci_user",
      "active": true,
      "createdOn": 1641839440620,
      "displayName": "Sam Andrus",
      "email": "sam.andrus@jupiterone.com",
      "mfaEnabled": false,
      "name": "Sam Andrus",
      "role": "owner",
      "status": "active",
      "username": "sam.andrus@jupiterone.com",
      "webLink": "https://dashboard.signalsciences.net/corps/jupiterone/users/sam.andrus@jupiterone.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "austin.kelleher@jupiterone.com",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "accessTokenAllowed": true,
            "accessTokens": Object {
              "uri": "/api/v0/user/apitokens",
            },
            "announcements": Object {
              "uri": "/api/v0/user/announcements",
            },
            "apiUser": false,
            "authStatus": "none",
            "corpAuthType": "builtin",
            "created": "2022-01-10T18:30:28.296Z",
            "defaultDashboards": Object {
              "uri": "/api/v0/user/defaultDashboards",
            },
            "email": "austin.kelleher@jupiterone.com",
            "memberships": Object {
              "data": Array [],
              "uri": "/api/v0/corps/jupiterone/users/austin.kelleher@jupiterone.com/memberships",
            },
            "mfaEnabled": false,
            "name": "Austin Kelleher",
            "role": "owner",
            "ssoProvisioned": false,
            "status": "active",
          },
        },
      ],
      "_type": "sigsci_user",
      "active": true,
      "createdOn": 1641839428296,
      "displayName": "Austin Kelleher",
      "email": "austin.kelleher@jupiterone.com",
      "mfaEnabled": false,
      "name": "Austin Kelleher",
      "role": "owner",
      "status": "active",
      "username": "austin.kelleher@jupiterone.com",
      "webLink": "https://dashboard.signalsciences.net/corps/jupiterone/users/austin.kelleher@jupiterone.com",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jupiterone",
      "_key": "jupiterone|has|samuel.poulton+sigsci_observer@jupiterone.com",
      "_toEntityKey": "samuel.poulton+sigsci_observer@jupiterone.com",
      "_type": "sigsci_corp_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jupiterone",
      "_key": "jupiterone|has|samuel.poulton@jupiterone.com",
      "_toEntityKey": "samuel.poulton@jupiterone.com",
      "_type": "sigsci_corp_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jupiterone",
      "_key": "jupiterone|has|sam.andrus@jupiterone.com",
      "_toEntityKey": "sam.andrus@jupiterone.com",
      "_type": "sigsci_corp_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jupiterone",
      "_key": "jupiterone|has|austin.kelleher@jupiterone.com",
      "_toEntityKey": "austin.kelleher@jupiterone.com",
      "_type": "sigsci_corp_has_user",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "sigsci_corp",
    "sigsci_user",
    "sigsci_corp_has_user",
  ],
  "numCollectedEntities": 5,
  "numCollectedRelationships": 4,
}
`;
